### ISSUE TITLE
Moderator flags can hide posts

### ISSUE THREAD

**Issue by docapotamus** [pjuu/pjuu#385]
We are seeing an increased amount of spam on https://pjuu.com.

I have spoken about this with the moderators and a good solution may be to have moderator flags increment a second counter.
Once this counter is > 2 we will hide the post.

Administrator access will still be required to delete the post.

**Comment by mcrmonkey** [comment]
:+1: 
Makes sense.  .. what about stages for the hiding ?
i.e. 2 for hide from global feed and 3 for site wide ?

**Comment by docapotamus** [comment]
@mcrmonkey sounds like a good plan. Most people won't be following these new accounts anyhow so the posts won't be in the feed.

**Comment by mcrmonkey** [comment]
no, but you could go find it either by the global feed or if someone mentions the user in a post

**Comment by docapotamus** [comment]
@mcrmonkey more people use the global feed, hiding it there is paramount

**Comment by mcrmonkey** [comment]
indeed - however you could still find that post if you found the user's feed

**Comment by docapotamus** [comment]
It shouldn't be too bad to do.

If when flagging a post, we check if the user is a moderator and increment a second flag 'mod_flags'. When we select the posts from MongoDB we guard all the queries that this flag has to be < 2.

This should work everywhere. The feed is just a list of IDs in Mongo which are then selected.
